### DISCUSSION TITLE
Rust bindings

### DISCUSSION THREAD

**Discussion by nolan-veed** [roc-streaming/roc-toolkit#707] [Discuss]
Hello folks,

Has this come up as a possible way to get roc-streaming into the rust eco-system?

Rust adoption is picking up, and it would be good to have a crate that gives us roc-streaming capabilities. What are the thoughts of a project around this.

Thanks,
Nolan

**Comment by gavv** [comment]
Hi,

Rust bindings are must-have, I agree. IIRC I even added it to the roadmap in sphinx some time ago, though nobody is working on it so far.

Personally, I haven't touched rust yet. I have an unread book on my shelf :)

I think @baranovmv has some experience with rust, and maybe he could do a review if someone will work on the bindings.

@nolan-veed in case if you're interested to work on them, you're welcome of course!

A starting point could be the script called generate-enums.py written by @ortex which we use to generate some boilerplate for roc-java and roc-go.

**Comment by Knoficooki** [comment]
I am actually interested in writing a wrapper since I want to use this library in Rust. I will be developing it on Codeberg ([link https://codeberg.org/Noffie/roc-rs.git](https://codeberg.org/Noffie/roc-rs.git)) and if I am happy with it I will try to maintain it alongside of rog-toolkit.
That means I will start with version 0.4.0 (if older version are needed, a merge request with the versions code (or any help) will be appreciated

**Reply by baranovmv** [reply]
That will be very cool. Thanks for sharing your plans. If you are interested we'll be happy to contribute by review, testing etc. Please consider to share api surface overview as early as possible in order to agree on terms and concepts with roc-streaming public api docs, so that struct, method and parameter naming will align with C API.

Again, your efforts will be very much appreciated

**Reply by Knoficooki** [reply]
The repository is now online. I will open an issue about the API structure, this will be the main place where discussions for the API will take place (until I find a feature that is actually meant for this)
